### PR TITLE
#268 Return isFavorite flag with Animal object

### LINKED ISSUE #268: Return isFavorite flag with Animal object
Check if an animal is marked as favorite by the current user and return a flag `isFavorite` with `Animal` object.

### PR DESCRIPTION
**Pull request by mantasimkus**
Supplemented `Animal` object with `isFavorite` flag which determines if an animal is marked as favorite by the current user. Resolves #268.

### CODE REVIEW

**Review by AzisK — APPROVED** [review]
Maybe I would join the animal table with favorite table but other than that it look ok

**Comment by mantasimkus** [comment]
@AzisK, as you noticed, currently `isFavorite` field is returned with a separate query (just like animal `registration`, `microchip`, and `details`). Are you suggesting to not follow the previous examples and join tables inside `getAnimal(s)Query`, or sth else?

**Comment by AzisK** [comment]
I am thinking that this approach might be faster. However, as you pointed out, we have previous examples and it would be good to follow the same path. So I just gave food for thought but go ahead with this approach

File: src/schema/resolvers/animal.ts [hunk 1]
```diff
@@ -325,6 +326,25 @@ const resolvers: IResolvers = {
...
+            }
+            
+            const dbResponse = await pgClient.query(
+                getFavoriteAnimalQuery({
+                    userId,
+                    animalId: id
+                })
+            );
+			
+            if (dbResponse.rows[0]) {
```
Comment (JustasBelevicius) [inline src/schema/resolvers/animal.ts:343]: could be: `return !!dbResponse.rows[0]`

**Review by JustasBelevicius — APPROVED** [review]
I agree with @AzisK, it probably would be nicer to just extend the animal query and model by adding the favourite flag (add logic to SQL - if user undefined/null, favourite always false). I believe an additional request for each animal is a bit unnecessary.

**Comment by mantasimkus** [comment]
OK, as both of you suggested, I extended `getAnimal(s)` queries with `isFavorite` status. However, a few tests do not pass anymore due to undefined `user_id` used in the queries mentioned before. Would you suggest to hardcode a dummy `user_id` in tests?

**Comment by JustasBelevicius** [comment]
Well done! Looks good from my point of view. @AzisK is this okay to merge for you?

**Comment by AzisK** [comment]
Looks great

**Pull request merged**